test/AutoRest.Shared.Tests/TypeFormatterTests.cs

```diff
@@ -55,5 +55,26 @@ public void FormatsDatesAsJson(string format, DateTimeOffset date, string expect
             Assert.AreEqual(expected, formatted.ToString());
             Assert.AreEqual(date, formatted.GetDateTimeOffset(format));
         }
+
+        [TestCase("2020-05-04T03:02:01.1230000+08:00")]
+        [TestCase("2020-05-04T03:02:01.1230000-08:00")]
+        [TestCase("2020-05-04T03:02:01.1230000+00:00")]
+        [TestCase("2020-05-04T03:02:01.1230000")]
+        [TestCase("Mon, 04 May 2020 03:02:01 GMT")]
+        [TestCase("Mon, 04 May 2020 03:02:01")]
+        public void TestEqualAfterConvertingToUtc(string dateString)
+        {
+            string[] formats = { "O", "o" };
+
+            foreach (string format in formats)
+            {
+                var originalDate = DateTimeOffset.Parse(dateString);
+                var originalTimeMillis = originalDate.ToUnixTimeMilliseconds();
+
+                var formatted = TypeFormatters.ToString(originalDate, format);
+                var utcDate = DateTimeOffset.Parse(formatted);
```

Comment:
Do we use the DateTimeOffset.Parse method when deserializing, or are we using the TypeFormatter? If we use the TypeFormatter then I think we should test with that as well..

Reply:
AFAIK, we never deserialize datetime by itself, it is either a part of a json or xml. For json, we call `Value.GetDateTimeOffset("O")`, for xml, we call `XElementExtensions.GetDateTimeOffsetValue`

I've checked both yesterday, both produced no errors.

Reply:
TypeFormatter ends up calling DateTimeOffset.Parse [here](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/TypeFormatters.cs#L130).  So I think we are good.